### PR TITLE
enh(openid): Extend PUT API to save redirect uri

### PR DESCRIPTION
**Pull request by dmyios**
## Description

Added PUT endpoint to save redirect URI for 22.04

Backport of https://github.com/centreon/centreon/pull/1200 to 22.04

**Fixes** # MON-18052

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [x] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Comment by jeremyjaouen** [comment]
please add the Update script

File: centreon/src/Core/Security/Application/ProviderConfiguration/OpenId/UseCase/UpdateOpenIdConfiguration/UpdateOpenIdConfigurationRequest.php [hunk 1]
```diff
@@ -130,6 +130,11 @@ class UpdateOpenIdConfigurationRequest
      */
     public ?string $userNameBindAttribute = null;
 
+    /**
+     * @var string|null
+     */
```
Comment (abuathier) [inline centreon/src/Core/Security/Application/ProviderConfiguration/OpenId/UseCase/UpdateOpenIdConfiguration/UpdateOpenIdConfigurationRequest.php:135]: Code style
```suggestion
    /** @var string|null */
```

File: centreon/src/Core/Security/Domain/ProviderConfiguration/OpenId/Model/Configuration.php [hunk 2]
```diff
@@ -149,6 +149,11 @@ class Configuration implements ProviderConfigurationInterface
      */
     private ?string $userNameBindAttribute = null;
 
+    /**
+     * @var string|null
+     */
```
Comment (abuathier) [inline centreon/src/Core/Security/Domain/ProviderConfiguration/OpenId/Model/Configuration.php:154]:
```suggestion
    /** @var string|null */
```

File: centreon/www/install/php/Update-22.04.13.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            FROM provider_configuration
+        WHERE
+            name = 'openid'
+        SQL
+    )->fetchColumn();
+
+    $customConfiguration = json_decode($customConfigurationJson, true);
+    if (! array_key_exists('redirect_url', $customConfiguration)) {
+        $customConfiguration['redirect_url'] = null;
+        $updatedCustomConfigurationEncoded = json_encode($customConfiguration);
```
Comment (abuathier) [inline centreon/www/install/php/Update-22.04.13.php:43]: Harden `json_encode` / `json_decode` with `JSON_THROW_ON_ERROR` ?

File: centreon/www/install/php/Update-22.04.13.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+        $statement = $pearDB->prepare(
+            <<<SQL
+            UPDATE provider_configuration
+                SET custom_configuration = :encodedConfiguration
+            SQL
```
Comment (abuathier) [inline centreon/www/install/php/Update-22.04.13.php:49]: Code style
```suggestion
                UPDATE provider_configuration
                    SET custom_configuration = :encodedConfiguration
                SQL
```

File: centreon/www/install/php/Update-22.04.13.php [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+$updateOpenIdCustomConfiguration = function (CentreonDB $pearDB): void
+{
+    $customConfigurationJson = $pearDB->query(
+        <<<'SQL'
+        SELECT custom_configuration
+            FROM provider_configuration
+        WHERE
+            name = 'openid'
+        SQL
```
Comment (abuathier) [inline centreon/www/install/php/Update-22.04.13.php:37]: Code style
```suggestion
        <<<'SQL'
            SELECT custom_configuration
                FROM provider_configuration
            WHERE
                name = 'openid'
            SQL
```

File: centreon/www/install/php/Update-22.04.13.php (hunk 3 above)
Comment (dmyios) [reply centreon/www/install/php/Update-22.04.13.php:43]: it's a backport PR

**Review by abuathier — APPROVED** [review]

**Review by wtermellil — APPROVED** [review]

**Pull request merged**